include/pybind11/attr.h

```diff
@@ -10,6 +10,8 @@
 
 #pragma once
 
+#include <functional>
+
 #include "cast.h"
```

Comment:
```suggestion
#include "cast.h"

#include <functional>
```

Pybind11 headers must always be included before stdlib.

Reply:
@Skylion007 is there a way to force this in `clang-format/tidy`?

Reply:
Thanks, done.